modules/tinymce/src/themes/silver/main/ts/ui/promotion/Promotion.ts

```diff
@@ -0,0 +1,45 @@
+import { AlloySpec, SimpleSpec } from '@ephox/alloy';
+
+import Editor from 'tinymce/core/api/Editor';
+
+import { UiFactoryBackstageShared } from '../../backstage/Backstage';
+
+const promotionMessage = 'Try Premium!';
+const promotionLink = 'http://tiny.cloud';
+
+export interface PromotionSpec extends SimpleSpec {
+  editor: Editor;
+  sharedBackstage: UiFactoryBackstageShared;
+}
+
+const renderPromotion = (spec: PromotionSpec): AlloySpec => {
+  return {
+    uid: spec.uid,
+    dom: spec.dom,
+
+    components: [
+      {
+        dom: {
+          tag: 'div',
+          classes: [ 'tox-promotion' ],
```

Comment:
This doesn't look right... won't this result in 2 `tox-promotion` divs being created? We should either be merging this with the `spec.dom` logic instead I believe or just not include it.

Reply:
I was going to check that, but forgot. My bad.